python/paddle/fluid/compiler.py

```diff
@@ -535,10 +704,39 @@ def __init__(self):
             self._ipu_strategy.set_options(default_options)
             self.has_custom_ops = False
             self.custom_op_names = []
+            self.need_compile = True
         else:
             raise RuntimeError(
                 "Can not use IpuStrategy in non IPU compiled environment, please re-compile with WITH_IPU=ON."
             )
+        from paddle import in_dynamic_mode
+        if in_dynamic_mode():
+            IpuDynamicPatcher.patch_program_cache(self)
+            IpuDynamicPatcher.patch_lr_scheduler(self)
```

Comment:
The main idea here is to convert the paddle program to IPU-supported program. And yes, it may introduce the error if user want to use `to_static` normally somewhere.

Will consider an registration here instead of direct patch.

Reply:
Added apis to register/release the patch.